app/src/main/java/org/fossasia/susi/ai/login/WelcomeActivity.java

```diff
@@ -0,0 +1,186 @@
+package org.fossasia.susi.ai.login;
+
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.design.widget.TabLayout;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.Button;
+
+import org.fossasia.susi.ai.R;
+
+/**
+ * Created by Arshil on 21/08/17.
+ */
+
+public class WelcomeActivity extends AppCompatActivity {
+
+    private ViewPager mImageViewPager;
+    private int[] layouts;
+    private Button btnSkip, btnNext;
+    private PrefManager prefManager;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Checking for first time launch - before calling setContentView()
+        prefManager = new PrefManager(this);
+        if (!prefManager.isFirstTimeLaunch()) {
+            launchHomeScreen();
+            finish();
+        }
+
+
+        // Making notification bar transparent
+        if (Build.VERSION.SDK_INT >= 21) {
+            getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN);
+        }
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
+                WindowManager.LayoutParams.FLAG_FULLSCREEN);
+        setContentView(R.layout.activity_welcome);
+
+        mImageViewPager = (ViewPager) findViewById(R.id.pager);
+        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabDots);
+        tabLayout.setupWithViewPager(mImageViewPager, true);
+
+
+
+        btnSkip = (Button) findViewById(R.id.btn_skip);
+        btnNext = (Button) findViewById(R.id.btn_next);
+
+
+        // layouts of all welcome sliders
+        layouts = new int[]{
+                R.layout.welcome_slide1,
+                R.layout.welcome_slide2,
+                R.layout.welcome_slide3,
+                R.layout.welcome_slide4};
+
+        // making notification bar transparent
+        changeStatusBarColor();
+        MyViewPagerAdapter myViewPagerAdapter = new MyViewPagerAdapter();
+        mImageViewPager.setAdapter(myViewPagerAdapter);
+        mImageViewPager.addOnPageChangeListener(viewPagerPageChangeListener);
+
+        btnSkip.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                launchHomeScreen();
+            }
+        });
+
+        btnNext.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                // checking for last page
+                // if last page home screen will be launched
+                int current = getItem(+1);
```

Comment:
What does `+1` in `int current = getItem(+1);` denote?

Reply:
+1  (which is written to be more precise rather than 1)  because the  getItem method  receives the +1 as an argument and returns  the next viewpager and then btnNext checks if there is a next viewpager or not, if not then it simply launched the LoginScreen.